### PR TITLE
refactor: typescript

### PR DESCRIPTION
**Pull request by menduz**
BREAKING CHANGES

- Instead of defining things in static properties and accessing them by `this.constructor.someStaticThing`, We follow the abstract class pattern to enable type checking, it happens mostly in the `Contract` class
- Look for `TODO: this unlock account method doesn't exist eth.unlockAccount()`
- It adds automatic typings to the projects installing this and using VSCode and/or typescript
- Most important, we no longer require babel, that requires babel-polyfill to be installed in the web app/node

### CODE REVIEW

File: scripts/contracts/IndexFile.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+    }
+  }
+}
+
 export class IndexFile {
-  constructor(contractPaths) {
-    this.contractPaths = contractPaths
+  contracts: ContractFile[]
+
+  constructor(public contractPaths: string[]) {
```
Comment (eordano) [inline scripts/contracts/IndexFile.ts]: Unused variable

[the remark this answers is not in the document]
Comment (menduz) [reply scripts/contracts/IndexFile.ts]: Nope, the `public` modifier creates a property with that name and signature.

You could do 

```ts
class Vertex {
  constructor(public x: number, public y: number, public z: number){}
}

V = new Vertex(1,2,3)
V.x == 1
```

**Comment by menduz** [comment]
@cazala done

**Comment by NicoSantangelo** [comment]
New lib.
Tracked by https://github.com/decentraland/decentraland-eth

**Pull request closed without merging**